herbert_core/utilities/maths/noisify_test_rand.m

```diff
@@ -0,0 +1,53 @@
+function vals = noisify_test_rand(n)
+%--------------------------------------------------
+% This function creates a simple deterministic series
+% of numbers which can be substituted for a true pseudo-random
+% number generator to enable by-eye checking of the random
+% noise addition (via noisify( to the signal. It can
+% substitute for randn in the Herbert noisify method.
+%
+% This function is designed to accumulate the "pseudo-
+% random" numbers across multiple calls - one for each
+% page of a paged sqw object. An external global flag
+% initialises these numbers, and an internal persistent
+% number keeps track of where this process has got to.
+%
+% Input:
+%     n - size of the signal vector for which the noise
+%         is being generated.
+% The global flag is expected to be acceptable in a test
+% though not in production code.
+%--------------------------------------------------
+
+% external global flag to signal that this pseudo-random
+% number distribution needs to be initialised if true.
+global noisify_test_rand_init;
+% internal number tracking where in the generation of
+% the sequence we have got to.
+persistent got_to;
```

Comment:
is there any way to reset this?

If there are multiple tests using this test function the results will vary depending on whether coupled tests have been run

Reply:
this function now recast as class object with latest commit

Reply:
and moved to _test/shared